distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso

```diff
@@ -142,3 +144,32 @@ resolve_aggregate table problem_builder aggregate_column =
 ## PRIVATE
    A marker for missing columns during resolution.
 type Internal_Missing_Column_Error
+
+## PRIVATE
+   Creates a Java AggregateColumnDefinition for the Aggregate_Column
+java_aggregator : Aggregate_Column->AggregateColumnDefinition
+java_aggregator name column =
+    case column of
+        Group_By c _ -> AggregateColumnDefinition.GroupBy name c.java_column
+        Count _ -> AggregateColumnDefinition.Count name
+        Count_Distinct columns _ ignore_nothing ->
+            resolved = columns.map .java_column
+            AggregateColumnDefinition.CountDistinct name resolved.to_array ignore_nothing
+        Count_Not_Nothing c _ -> AggregateColumnDefinition.CountNothing name c.java_column False
+        Count_Nothing c _ -> AggregateColumnDefinition.CountNothing name c.java_column True
+        Count_Not_Empty c _ -> AggregateColumnDefinition.CountEmpty name c.java_column False
+        Count_Empty c _ -> AggregateColumnDefinition.CountEmpty name c.java_column True
+        Sum c _ -> AggregateColumnDefinition.Sum name c.java_column
+        Average c _ -> AggregateColumnDefinition.Mean name c.java_column
+        Standard_Deviation c _ p -> AggregateColumnDefinition.StDev name c.java_column p
+        Median c _ -> AggregateColumnDefinition.Percentile name c.java_column 0.5
+        Percentile p c _ -> AggregateColumnDefinition.Percentile name c.java_column p
+        Mode c _ -> AggregateColumnDefinition.Mode name c.java_column
+        First c _ ignore_nothing _ -> AggregateColumnDefinition.First name c.java_column ignore_nothing
+        Last c _ ignore_nothing _ -> AggregateColumnDefinition.Last name c.java_column ignore_nothing
+        Maximum c _ -> AggregateColumnDefinition.MinOrMax name c.java_column 1
+        Minimum c _ -> AggregateColumnDefinition.MinOrMax name c.java_column -1
+        Shortest c _ -> AggregateColumnDefinition.MinOrMaxLength name c.java_column -1
+        Longest c _ -> AggregateColumnDefinition.MinOrMaxLength name c.java_column 1
+        Concatenate c _ join prefix suffix quote -> AggregateColumnDefinition.Concatenate name c.java_column join prefix suffix quote
+        _ -> Error.throw "Unsupported aggregation."
```

Comment:
I think we shouldn't throw pure `Text` values. This will render badly in the GUI.

I know the codebase still has some places like this, but whenever we modify something we should go in the direction of avoiding this. I think `Illegal_Argument_Error` or `Illegal_State_Error` are the universal wrappers to use, or in this case maybe it's worth creating a separate class? Like the `Unsupported_Database_Operation_Error` we were talking about.